demos/projects/ESPRESSIF/nvs-rw-ca-cert/main/nvs_blob_example_main.c

```diff
@@ -0,0 +1,282 @@
+#endif
+
+/* Read from NVS and print restart counter
+   and the table with run times.
+   Return an error if anything goes wrong
+   during this process.
+ */
+esp_err_t print_what_saved(void)
+{
+    nvs_handle_t my_handle;
+    esp_err_t err;
+
+    // Open
+    err = nvs_open(CA_CERT_NAMESPACE, NVS_READWRITE, &my_handle);
+    if (err != ESP_OK)
+    {
+      return err;
+    }
+
+    // Read AZURE_TRUST_BUNDLE_VERSION_NAME
+    int32_t trust_bundle_version = 0; // value will default to 0, if not set yet in NVS
+    err = nvs_get_i32(my_handle, AZURE_TRUST_BUNDLE_VERSION_NAME, &trust_bundle_version);
+    if (err != ESP_OK && err != ESP_ERR_NVS_NOT_FOUND)
+    {
+      return err;
+    }
+    printf("Trust bundle version = %d\n", trust_bundle_version);
+
+    // Read run time blob
+    size_t trust_bundle_read_size = 0;  // value will default to 0, if not set yet in NVS
+    // obtain required memory space to store blob being read from NVS
+    err = nvs_get_blob(my_handle, AZURE_TRUST_BUNDLE_NAME, NULL, &trust_bundle_read_size);
+    if (err != ESP_OK && err != ESP_ERR_NVS_NOT_FOUND)
+    {
+      return err;
+    }
+
+    if (trust_bundle_read_size == 0)
+    {
+        printf("Nothing saved yet!\n");
+    }
+    else
+    {
+        uint8_t* trust_bundle_read = malloc(trust_bundle_read_size);
+        err = nvs_get_blob(my_handle, AZURE_TRUST_BUNDLE_NAME, trust_bundle_read, &trust_bundle_read_size);
+
+        if (err != ESP_OK)
+        {
+            free(trust_bundle_read);
+            return err;
+        }
+
+        printf("Stored trust bundle:\n");
+        printf("%.*s\n", trust_bundle_read_size, trust_bundle_read);
+
+        free(trust_bundle_read);
+    }
+
+    // Close
+    nvs_close(my_handle);
+
+    return ESP_OK;
+}
+
+/* Save new run time value in NVS
+   by first reading a table of previously saved values
+   and then adding the new value at the end of the table.
+   Return an error if anything goes wrong
+   during this process.
+ */
+esp_err_t save_trust_bundle(void)
+{
+    nvs_handle_t my_handle;
+    esp_err_t err;
+
+    // Open CA Cert namespace
+    err = nvs_open(CA_CERT_NAMESPACE, NVS_READWRITE, &my_handle);
+    if (err != ESP_OK)
+    {
+      return err;
+    }
+
+    // Read the current trust bundle version
+    int32_t read_trust_bundle_version = 0;  // value will default to 0, if not set yet in NVS
+    err = nvs_get_i32(my_handle, AZURE_TRUST_BUNDLE_VERSION_NAME, &read_trust_bundle_version);
+    if (err != ESP_OK && err != ESP_ERR_NVS_NOT_FOUND)
+    {
+      return err;
+    }
+
+    // If version matches, do not write to not overuse NVS
+    if (read_trust_bundle_version == trust_bundle_version)
+    {
+      printf("Trust bundle version in NVS matches bundle version to write.\n");
+      return ESP_OK;
+    }
+
+    // Set new trust bundle version
+    printf("Writing trust bundle version\n");
+    err = nvs_set_i32(my_handle, AZURE_TRUST_BUNDLE_VERSION_NAME, trust_bundle_version);

```

Comment:
do this after the blob write just in case the device reboots or losses power before the bundle is written.

Reply:
or does nvs_commit take care of this.